src/com/t_oster/liblasercut/drivers/Grbl.java

```diff
@@ -176,38 +271,60 @@ protected void move(PrintStream out, double x, double y, double resolution) thro
     x = isFlipXaxis() ? getBedWidth() - Util.px2mm(x, resolution) : Util.px2mm(x, resolution);
     y = isFlipYaxis() ? getBedHeight() - Util.px2mm(y, resolution) : Util.px2mm(y, resolution);
     currentSpeed = getTravel_speed();
+
+    String line = "G0";
+
+    if (resolution >= 254) {
+      line += String.format(FORMAT_LOCALE, "X%.3fY%.3f",x,y);
+    } else {
+      line += String.format(FORMAT_LOCALE, "X%.2fY%.2f",x,y);
+    }
+
+    currentX = x;
+    currentY = y;
+
     if (blankLaserDuringRapids)
     {
       currentPower = 0.0;
-      sendLine("G0 X%f Y%f S0", x, y);
-    }
-    else
-    {
-      sendLine("G0 X%f Y%f", x, y);
+      line += "S0";
     }
+
+    sendLine(line);
+
   }
-  
+
   /**
    * Send a line of gcode to the cutter, stripping out any whitespace in the process
    * @param text
    * @param parameters
    * @throws IOException 
    */
   @Override
+
   protected void sendLine(String text, Object... parameters) throws IOException
   {
-    out.format(FORMAT_LOCALE, text.replace(" ", "")+LINEEND(), parameters);
-    //TODO: Remove
-    System.out.println(String.format(FORMAT_LOCALE, "> "+text+LINEEND(), parameters));
-    out.flush();
-    if (isWaitForOKafterEachLine())
-    {
-      String line = waitForLine();
-      if (!"ok".equals(line))
-      {
-        throw new IOException("Lasercutter did not respond 'ok', but '"+line+"'instead.");
-      }
-    }
+    if (isSimpleStreamMode())
+      sendLineSimple(text, parameters);
+    else
+      sendLineCC(text, parameters);
+  }
+
+  @Override
+  protected void sendJobPrepare() throws IOException {
+    setSimpleStreamMode(false);
+  }
+
+  @Override
+  protected void sendJobFinish() throws IOException {
+    setSimpleStreamMode(true);
+  }
+
```

Comment:
You need to set and clear it before and after the job? Why? Does the initializationcode have to be send in simplestreammode? I do not quite get this mode, but because I do not have any GRBL compatible device, I cannot test it anyway, but I guess you know what you are doing.
However, if it can be enabled all the time, it would be easy to make this property configurable in the driver's settings. I don't know if there are any GRBL devices, which do not understand that mode.